### PR TITLE
feat(config-server): add MongoDB environment repository support

### PR DESCRIPTION
**Pull request by apappascs**
This commit introduces MongoDB as a new backend option for the Spring Cloud Config Server, 
enabling users to store and manage their configuration properties in a MongoDB database.

### CODE REVIEW

File: spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/MongoDbEnvironmentRepository.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+	public MongoDbEnvironmentRepository(MongoTemplate mongoTemplate, MongoDbEnvironmentProperties properties) {
+		this.mongoTemplate = mongoTemplate;
+		this.properties = properties;
+	}
+
+	@Override
+	public Environment findOne(String application, String profile, String label) {
+		label = Optional.ofNullable(label).filter(StringUtils::hasText).orElse(this.properties.getDefaultLabel());
+		profile = Optional.ofNullable(profile).filter(StringUtils::hasText).orElse("default");
```
Comment (spencergibb) [inline spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/MongoDbEnvironmentRepository.java:58]: We don't use Optional except for certain return and parameter situations. Please use the ternary operator

File: spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/MongoDbEnvironmentRepositoryTests.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+@ActiveProfiles("mongodb")
+@Testcontainers
+@Tag("DockerRequired")
+public class MongoDbEnvironmentRepositoryTests {
+
+	@Container
+	public static GenericContainer<?> mongoContainer = new GenericContainer<>("mongo:6.0.5").withExposedPorts(27017);
+
+	@DynamicPropertySource
```
Comment (spencergibb) [inline spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/MongoDbEnvironmentRepositoryTests.java:66]: Is Mongo one of the supported dbs with @ServiceConnection?

**Review by spencergibb — COMMENTED** [review]
A few minor requests after a cursory review

File: docs/modules/ROOT/pages/server/environment-repository/mongo-backend.adoc [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+		</dependency>
+	</dependencies>
+----
+
+Configure your application's `application.properties` or `application.yml` to point to your MongoDB instance:
+
+[source,yaml]
+----
+spring:
+  data:
```
Comment (ryanjbaxter) [inline docs/modules/ROOT/pages/server/environment-repository/mongo-backend.adoc:24]: The profile `mongodb` would need to be enabled as well

**Comment by ryanjbaxter** [comment]
Thanks for the PR!  We are going to wait to merge this until we branch for our 2024 release train later this year.  Once we branch we will get this merged.

**Pull request merged**